packages/engine/src/networking/serialization/DataWriter.ts

```diff
@@ -65,172 +67,172 @@ export const writeComponent = (component: any, periodicUnchanged = false) => {
+  const rewind = rewindViewCursor(v)
+  const writeChangeMask = spaceUint8(v)
+  let changeMask = 0
+  let b = 0
 
-    changeMask |= writePropIfChanged(v, vector3.x, entity, ignoredChanged) ? 1 << b++ : b++ && 0
-    changeMask |= writePropIfChanged(v, vector3.y, entity, ignoredChanged) ? 1 << b++ : b++ && 0
-    changeMask |= writePropIfChanged(v, vector3.z, entity, ignoredChanged) ? 1 << b++ : b++ && 0
+  const ignoreHasChanged = hasComponent(entity, NetworkObjectSendPeriodicUpdatesTag)
 
-    return (changeMask > 0 && writeChangeMask(changeMask)) || rewind()
-  }
+  changeMask |= writePropIfChanged(v, vector3.x, entity, ignoreHasChanged) ? 1 << b++ : b++ && 0
+  changeMask |= writePropIfChanged(v, vector3.y, entity, ignoreHasChanged) ? 1 << b++ : b++ && 0
+  changeMask |= writePropIfChanged(v, vector3.z, entity, ignoreHasChanged) ? 1 << b++ : b++ && 0
+
+  return (changeMask > 0 && writeChangeMask(changeMask)) || rewind()
+}
 
 // Writes a compressed Vector3 value to the DataView.
-export const writeCompressedVector3 =
-  (vector3: Vector3SoA) =>
-  (v: ViewCursor, entity: Entity, ignoredChanged = false) => {
-    const rewind = rewindViewCursor(v)
-    const writeChangeMask = spaceUint8(v)
-    const rewindUptoChageMask = rewindViewCursor(v)
-    let changeMask = 0
-    let b = 0
+export const writeCompressedVector3 = (vector3: Vector3SoA) => (v: ViewCursor, entity: Entity) => {
+  const rewind = rewindViewCursor(v)
+  const writeChangeMask = spaceUint8(v)
+  const rewindUptoChageMask = rewindViewCursor(v)
+  let changeMask = 0
+  let b = 0
 
-    changeMask |= writePropIfChanged(v, vector3.x, entity, ignoredChanged) ? 1 << b++ : b++ && 0
-    changeMask |= writePropIfChanged(v, vector3.y, entity, ignoredChanged) ? 1 << b++ : b++ && 0
-    changeMask |= writePropIfChanged(v, vector3.z, entity, ignoredChanged) ? 1 << b++ : b++ && 0
+  const ignoreHasChanged = hasComponent(entity, NetworkObjectSendPeriodicUpdatesTag)
 
-    if (changeMask > 0) {
-      rewindUptoChageMask()
-      let [x, y, z] = [vector3.x[entity], vector3.y[entity], vector3.z[entity]]
+  changeMask |= writePropIfChanged(v, vector3.x, entity, ignoreHasChanged) ? 1 << b++ : b++ && 0
+  changeMask |= writePropIfChanged(v, vector3.y, entity, ignoreHasChanged) ? 1 << b++ : b++ && 0
+  changeMask |= writePropIfChanged(v, vector3.z, entity, ignoreHasChanged) ? 1 << b++ : b++ && 0
 
-      // Since avatar velocity values are too small and precison is lost when quantized
-      let offset_mult = 1
-      if (hasComponent(entity, AvatarComponent)) offset_mult = 100
+  if (changeMask > 0) {
+    rewindUptoChageMask()
+    let [x, y, z] = [vector3.x[entity], vector3.y[entity], vector3.z[entity]]
 
-      x *= VEC3_MAX_RANGE * VEC3_PRECISION_MULT * offset_mult
-      y *= VEC3_MAX_RANGE * VEC3_PRECISION_MULT * offset_mult
-      z *= VEC3_MAX_RANGE * VEC3_PRECISION_MULT * offset_mult
+    // Since avatar velocity values are too small and precison is lost when quantized
```

Comment:
If the quantization and low precision is affecting avatars, wouldn't it be affecting other objects as well?